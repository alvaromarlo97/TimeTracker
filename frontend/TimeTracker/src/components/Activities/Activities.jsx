/* eslint-disable no-underscore-dangle */
/* eslint-disable global-require */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-bind */
import React from 'react';
import {
  SafeAreaView,
  ScrollView,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { loadCurrentActivity, deleteActivity } from '../../redux/actions/userCreators';
import styles from './activities.style';

export default function Activities({ navigation }) {
  const dispatch = useDispatch();

  function setActivity(activityId) {
    dispatch(loadCurrentActivity({
      currentActivityId: activityId,
    }));

    navigation.navigate('MyTimer');
  }
  const currentUserId = useSelector(({ loggedUser }) => loggedUser?.user?._id);
  const activities = useSelector(({ loggedUser }) => loggedUser?.user?.activities);
  function removeActivity(activityId) {
    dispatch(deleteActivity(currentUserId, {
      activities: [activityId],
    }));
  }
  return (
    <SafeAreaView style={styles.container}>

      <ScrollView style={styles.scroll} showsVerticalScrollIndicator={false}>
        {activities.map((element) => (
          <>
            <TouchableOpacity
              style={styles.button}
              onPress={() => setActivity(element._id)}
              key={element._id}
            >
              <View style={[styles.x, { backgroundColor: element.color }]} />
              <Text key={element}>
                {element.activityName}
              </Text>
              <TouchableOpacity
                style={styles.x}
                onPress={() => removeActivity(element._id)}
              />
            </TouchableOpacity>
          </>
        ))}
      </ScrollView>
      <TouchableOpacity
        style={styles.addActivity}
        onPress={() => navigation.push('CreateAct')}
      >
        <Text>Add Activity</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
}
