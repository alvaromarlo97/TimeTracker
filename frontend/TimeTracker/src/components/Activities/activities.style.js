import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(235,255,254,1)',
    borderRadius: 25,
    display: 'flex',
    alignItems: 'center',
  },
  image: {
    marginTop: 100,
    width: 200,
    height: 200,
  },
  email: {
    fontFamily: 'arial',
    fontSize: 25,
    color: '#121212',
    height: 69,
    width: 263,
    backgroundColor: 'rgba(196,255,252,1)',
    textAlign: 'center',
    borderRadius: 25,
    marginTop: 100,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  email1: {
    fontFamily: 'arial',
    fontSize: 25,
    color: '#121212',
    height: 69,
    width: 263,
    backgroundColor: 'rgba(196,255,252,1)',
    textAlign: 'center',
    borderRadius: 25,
    marginTop: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  button: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: 350,
    height: 69,
    backgroundColor: 'rgba(30,191,178,1)',
    borderRadius: 25,
    marginTop: 25,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  login: {
    fontFamily: 'arial',
    color: '#121212',
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 30,
    marginTop: 8,
  },
});
