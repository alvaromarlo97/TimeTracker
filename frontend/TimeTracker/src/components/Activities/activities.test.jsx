import React from 'react';
import { render } from '../../../utils/testUtils/test.utils';
import Activities from './Activities';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Character Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<Activities navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
