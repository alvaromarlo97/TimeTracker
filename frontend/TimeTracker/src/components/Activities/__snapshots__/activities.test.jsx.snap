// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#262155",
      "borderRadius": 0,
      "display": "flex",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 30,
        "width": "95%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FCFFFF",
          "fontFamily": "Avenir-Medium",
          "fontSize": 50,
          "marginLeft": 20,
        }
      }
    >
      MyActivities
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 50,
          "display": "flex",
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/public/image/logout.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "display": "flex",
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "borderBottomLeftRadius": 10,
        "borderTopLeftRadius": 50,
        "marginBottom": -50,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FD9F32",
        "borderRadius": 50,
        "display": "flex",
        "height": 100,
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": 20,
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FCFFFF",
          "fontFamily": "Avenir-Medium",
          "fontSize": 50,
        }
      }
    >
      +
    </Text>
  </View>
</RCTSafeAreaView>
`;
